src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8333,6 +8379,312 @@ private StatementSyntax ParseLocalDeclarationStatement()
+                    }
+
+                    var closeParen = this.EatToken(SyntaxKind.CloseParenToken);
+
+                    SyntaxToken equals = null;
+                    ExpressionSyntax expression = null;
+                    if (withEquals)
+                    {
+                        equals = this.EatToken(SyntaxKind.EqualsToken);
+                        if (equals.IsMissing)
+                        {
+                            return null;
+                        }
+
+                        expression = this.ParseExpressionCore();
+                    }
+
+                    var result = _syntaxFactory.VariableDeconstructionDeclarator(openParen, list, closeParen, equals, expression);
+
+                    if (!result.ContainsDiagnostics && list.Count < 2)
+                    {
+                        result = this.AddError(result, ErrorCode.ERR_DeconstructTooFewElements);
+                    }
+
+                    return result;
+                }
+                finally
+                {
+                    _pool.Free(list);
+                }
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private VariableDeclarationSyntax ParseDeconstructionIdentifierOrIdentifiersParts(bool withEquals)
+        {
+            if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken)
+            {
+                return _syntaxFactory.VariableDeclaration(ParseDeconstructionIdentifiers(withEquals: false));
+            }
+            else
+            {
+                return ParseDeconstructionIdPart();
+            }
+        }
+
+        /// <summary>
+        /// Parses identifiers in a deconstruction. For instance, `var (id, id) = ...`
+        /// </summary>
+        private VariableDeclarationSyntax ParseDeconstructionIdPart()
+        {
+            var identifier = ParseIdentifierToken();
+            var declarator = _syntaxFactory.VariableDeclarator(identifier, null, null);
+            return _syntaxFactory.VariableDeclaration(declarator);
+        }
+
+        /// <summary>
+        /// Parses a type and identifier in a deconstruction. For instance, `(type id, type id) = ...`
+        /// </summary>
+        private VariableDeclarationSyntax ParseDeconstructionTypeIdPart()
+        {
+            var type = ParseType(parentIsParameter: false);
+            var identifier = ParseIdentifierToken();
+
+            var declarator = _syntaxFactory.VariableDeclarator(identifier, null, null);
+            return _syntaxFactory.VariableDeclaration(type, declarator);
+        }
+
+        /// <summary>
+        /// Check ahead for a deconstruction declaration, which requires deconstruction-variables with at least one good-looking variable, followed by an equals sign.
+        /// Doesn't move the cursor.
+        /// PROTOTYPE(tuples) Can this be done without allocations?
+        /// </summary>
+        private bool IsPossibleDeconstructionDeclaration()
```

Comment:
No, `TryParseDeconstructionDeclaration` can move the cursor if it succeeds. But `IsPossibleDeconstructionDeclaration` doesn't move the cursor.